### PR TITLE
Support skygear.yaml explicitly

### LINKED ISSUE #184: Update skygear.yaml format
- [ ] Change `deployments` from hash to list

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #184

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -37,18 +36,6 @@ const configPaths: { [domain: string]: string } = {
   project: "./skygear.yaml",
 };
 
-function migrate(configObject: Dictionary<any>) {
```
Comment (kiootic) [inline src/config.ts:40]: I think we can keep this?

Comment (louischan-oursky) [reply src/config.ts:40]: If we want to help the developer to migrate their old skygear.yaml to a new one, it is better to send the old one to the controller to perform the migration.

Comment (kiootic) [reply src/config.ts:40]: I think `skygear.yaml` is a structure that exists entirely within `skycli`, and `skycli` translates `skygear.yaml` into appropriate controller API arguments?

Comment (louischan-oursky) [reply src/config.ts:40]: Should the create-deployment endpoint supports versioning as well? A particular copy of skycli uses a fixed version. When the developer wants to upgrade, they install a newer skycli. skycli has a separate command to migrate `skygear.yaml` locally.

Comment (kiootic) [reply src/config.ts:40]: I suppose so... that is we are going to version the 'deployment' resource.

Comment (louischan-oursky) [reply src/config.ts:40]: I will remove this function in this PR. ~~In a later PR for https://github.com/SkygearIO/skycli/issues/185 I am going to add it back.~~ The migration is for skycli config, not skygear.yaml

**Comment by louischan-oursky** [comment]
Ready for review.

**Review by kiootic — APPROVED** [review]

**Pull request merged**